vip-portal/src/main/resources/vm/wrapper.vm

```diff
@@ -19,6 +19,23 @@ function error {
   echo [ ERROR - $D ] $* >&2
 }
 
+function function_singularity {
+    cat > singularity << 'EOF'
+          #!/bin/bash
+          /cvmfs/dirac.egi.eu/container/apptainer/bin/apptainer "$@"
+EOF
+}
+
+# Call the function to create the singularity script and make it executable
+function_singularity
+chmod +x singularity
```

Comment:
as for udocker, better do a `chmod a+x`.
also, move this line inside the `function_singularity` method to change the rights just after you create it

Reply:
Updated to chmod a+x singularity